src/hera/workflows/script.py

```diff
@@ -555,8 +555,11 @@ def _extract_return_annotation_output(source: Callable) -> List:
     if get_workflow_annotation(return_annotation):
         output.append(annotation_args)
     elif origin_type is tuple:
-        for annotated_type in annotation_args:
-            output.append(get_args(annotated_type))
+        if all(get_workflow_annotation(annotated_type) for annotated_type in annotation_args):
+            for annotated_type in annotation_args:
+                output.append(get_args(annotated_type))
+        elif any(get_workflow_annotation(annotated_type) for annotated_type in annotation_args):
+            raise ValueError("All elements in the tuple should be annotated with Artifact/Parameter.")
```

Comment:
I think we should include the function name in the error message and explain that this is an error about the return type annotation

Reply:
Makes sense. Fixed!